lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -2336,6 +2336,305 @@ class ConvertCastEquivalentOp : public OpConversionPattern<OpTy> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenGridSamplerOp : public OpConversionPattern<AtenGridSamplerOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenGridSamplerOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Location loc = op->getLoc();
+    // calculate input dimensions
+    Value input = adaptor.getInput();
+    auto inputType = input.getType().cast<RankedTensorType>();
+    auto inputShape = inputType.getShape();
+    Value innerDim_0a =
```

Comment:
A bunch of the math below is materializing static values then using `arith` do constant propagation. For initially support require that the input types have static shapes (`RankedTensorType` has a `hasStaticShape()` function). Then we can worry about dynamic shapes in the future. this way we can shorten this initially.

This should substantially simplify this block initially.

Reply:
probably need to check this one again